python/repositories.bzl

```diff
@@ -564,16 +564,20 @@ def python_register_toolchains(
                 platform = platform,
             ))
 
-    toolchains_repo(
-        name = toolchain_repo_name,
+    toolchain_aliases(
```

Comment:
We need it for pip, which is coming next.

Reply:
I was wondering actually why we were still having the `use_repo` statements in the example `MODULE.bazel`. I would expect the users to specificy the python version as a parameter to the `pip.parse` and it should just work without me needing to download or setup any hub repos on my end.

Since the toolchains are registered by `rules_python` this should be possible, I think.

Reply:
We need some of the `use_repo` statements in the MODULE.bazel.  For instance, the user needs to access `@python:defs.bzl` in the BUILD files.

I would rather not change too much, to be honest.  This is a large refactor, and `pip` is even more complicated.  After more testing, I'm happy to file an issue and remove the code.

Reply:
> We need it for pip, which is coming next.

Huh? Why? Pip doesn't use toolchains.

Reply:
Why do we have it in the non-bzlmod code base?

Reply:
Ah, ok, I think I see now. The toolchains_aliases repo is generating the `defs.bzl` file for the version-aware wrapper macros. We have some usages of those in the bzlmod examples for testing.